include/af/vision.h

```diff
@@ -208,26 +208,34 @@ AFAPI void hammingMatcher(array& idx, array& dist,
 
 #if AF_API_VERSION >= 31
 /**
-   C++ Interface wrapper for Nearest Neighbour
-
-   \param[out] idx is an array of MxN size, where M is equal to the number of query
-               features and N is equal to n_dist. The value at position IxJ indicates
-               the index of the Jth smallest distance to the Ith query value in the
-               train data array.
-               the index of the Ith smallest distance of the Mth query.
-   \param[out] dist is an array of MxN size, where M is equal to the number of query
-               features and N is equal to n_dist. The value at position IxJ indicates
-               the distance of the Jth smallest distance to the Ith query value in the
-               train data array based on the dist_type chosen.
-   \param[in]  query is the array containing the data to be queried
-   \param[in]  train is the array containing the data used as training data
-   \param[in]  dist_dim indicates the dimension to analyze for distance (the dimension
-               indicated here must be of equal length for both query and train arrays)
-   \param[in]  n_dist is the number of smallest distances to return (currently only
-               values <= 256 are supported)
-   \param[in]  dist_type is the distance computation type. Currently \ref AF_SAD (sum
-               of absolute differences), \ref AF_SSD (sum of squared differences), and
-               \ref AF_SHD (hamming distances) are supported.
+   C++ interface wrapper for determining the nearest neighbouring points to a
+   given set of points
+
+   \param[out] idx       is an array of MxN size, where M is \p n_dist and N is
+                         the number of queries. The value at position i,j is the
+                         \p train points list index with the ith smallest
+                         distance to the jth \p query point.
+   \param[out] dist      is an array of MxN size, where M is \p n_dist and N is
+                         the number of queries. The value at position i,j is the
+                         distance of the ith nearest point in \p train to the
+                         jth \p query point. This distance is computed according
+                         on the \p dist_type chosen.
+   \param[in]  query     is the array containing the points to be queried. The
+                         points must be described along dim0 and listed along
+                         dim1 (vice versa if \p dist_dim is 1)
+   \param[in]  train     is the array containing the points used as training
+                         data. The points must be described along dim0 and
+                         listed along dim1 (vice versa if \p dist_dim is 1)
+   \param[in]  dist_dim  indicates the dimension along which each point is
+                         described in \p query and \p train (the dimension
+                         specified here must be the same for both \p query and
+                         \p train arrays)
```

Comment:
I think we should be saying something like - data layout should follow same pattern described by `dist_dim` in both query and train arrays. Isn't that right ?

Reply:
Yeah, the data layout should follow the same pattern in `query` and `train`, depending on what `dist_dim` is. We should make that clearer in this definition (and explaining what `dist_dim` is in general for this function's docs).